test/io/write_sidb_layout_to_svg_in_siqad_style.cpp

```diff
@@ -0,0 +1,466 @@
+//
+// Created by Jan Drewniok on 05.11.24.
+//
+
+#include <catch2/catch_template_test_macros.hpp>
+#include <catch2/catch_test_macros.hpp>
+
+#include "utils/version_info.hpp"
+
+#include <fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp>
+#include <fiction/technology/cell_technologies.hpp>
+#include <fiction/technology/charge_distribution_surface.hpp>
+#include <fiction/technology/sidb_charge_state.hpp>
+#include <fiction/types.hpp>
+
+#include <fmt/format.h>
+
+#include <algorithm>
+#include <cctype>
+#include <sstream>
+#include <string>
+
+using namespace fiction;
+
+/**
+ * This function takes an SVG string as input and returns a new string with all
+ * whitespace characters removed. The purpose is to create a compact representation
+ * of the SVG content, which can be useful for comparisons, optimizations, or transmission.
+ *
+ * @param svg The input SVG string to be normalized.
+ * @return A string with all whitespace characters removed.
+ */
+auto normalize_svg = [](const std::string& svg) noexcept
+{
+    std::string result = svg;
+
+    // Remove all whitespace (spaces, tabs, newlines)
+    result.erase(std::remove_if(result.begin(), result.end(), ::isspace), result.end());
+
+    return result;
+};
+
+const std::string EXPECTED_SVG_LIGHT_CELL_LEVEL = fmt::format(R"(<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```

Comment:
Same rules as above apply to all these global constants.

Reply:
@marcelwa, ``constexpr`` does not work here, right?

Reply:
Unfortunately, `std::string` cannot be `constexpr` indeed. If `fmt::format` supports it, however, you could try changing the type of this expression to `auto` (or `constexpr const char*`) and append a `.c_str()` to the `fmt::format` call. Not a guarantee that it works, but worth a shot :man_shrugging: